packages/fiori/src/Timeline.ts

```diff
@@ -111,6 +188,116 @@ class Timeline extends UI5Element {
 			: Timeline.i18nBundle.getText(TIMELINE_ARIA_LABEL);
 	}
 
+	get growsOnScroll(): boolean {
+		return this.growing === TimeLineGrowingMode.Scroll;
+	}
+
+	get timeLineEndDOM(): Element {
+		return this.shadowRoot!.querySelector(".ui5-time-line-end-marker")!;
+	}
+
+	get growingButtonIcon() {
+		return this.layout === TimelineLayout.Horizontal ? "process" : "drill-down";
+	}
+
+	get moreBtn(): HTMLElement | null {
+		const domRef = this.getDomRef();
+
+		if (this.growsWithButton && domRef) {
+			return domRef.querySelector<HTMLElement>(`#${this._id}-growingButton`);
+		}
+
+		return null;
+	}
+
+	get showBusyIndicatorOverlay() {
+		return !this.growsWithButton && this.loading;
+	}
+
+	get growsWithButton(): boolean {
+		return this.growing === TimeLineGrowingMode.Button;
+	}
+
+	onAfterRendering() {
+		if (this.growsOnScroll) {
+			this.observeTimeLineEnd();
+		}
+	}
+
+	onEnterDOM() {
+		this.growingIntersectionObserver = this.getIntersectionObserver();
+	}
+
+	onExitDOM() {
+		this.growingIntersectionObserver!.disconnect();
```

Comment:
I believe it would be a good idea to do a non-null check before disconnecting to avoid potetial runtime errors which could happen in different frameworks for example, like React, Vue, etc.

```ts
if (this.growingIntersectionObserver) {
		this.growingIntersectionObserver.disconnect();
		this.growingIntersectionObserver = null;
		this.timeLineEndObserved = false;
	}
```

Reply:
Done